modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -5088,14 +5106,16 @@ private void unlockListenerReadLock() {
 
     /** {@inheritDoc} */
     @Override public String toString() {
-        lockEntry();
-
-        try {
-            return S.toString(GridCacheMapEntry.class, this);
-        }
-        finally {
-            unlockEntry();
+        if (tryLockEntry(ENTRY_LOCK_TIMEOUT)) {
+            try {
+                return S.toString(GridCacheMapEntry.class, this);
+            }
+            finally {
+                unlockEntry();
+            }
         }
+        else
+            return EMPTY_STRING;
```

Comment:
Why totally empty string? It will look suspicious in logs.
I suggest to print final fields (at least, key) and add disclaimer that full info is not available due to lock timeout. Something like this:
GridCacheMapEntry [key=org.gridgain.gmc.dto.cache.CacheId [idHash=288058555, hash=-337430116, clusterId=38b26fe7-769d-4f9a-b6ad-7af4e11fd6dc, cacheName=PUBLIC], <full info is unavailable due to entry lock timeout>]